lib/bolt/node.rb

```diff
@@ -53,15 +43,27 @@ def initialize(host, port = nil, user = nil, password = nil, uri: nil,
       @orch_task_environment = transport_conf[:orch_task_environment]
       @extensions = transport_conf[:extensions]
 
-      @logger = Logger.get_logger(progname: @host)
-      @transport_logger = Logger.get_logger(progname: @host, log_level: Logger::WARN)
+      @logger = Logger.get_logger(progname: host)
+      @transport_logger = Logger.get_logger(progname: host, log_level: Logger::WARN)
+    end
+
+    def host
+      @target.host
+    end
+
+    def port
+      @target.port
+    end
+
+    def uri
+      @target.uri.to_s.sub(%r{^/+}, '')
```

Comment:
Choosing to not preserve the original uri string and instead try to generate it from an Addressable::URI has some interesting side-effects (if there's no protocol/scheme, it still adds a `//` prefix that we don't really want). I may go back to preserving the original uri string separately.